plugin/bufsurf.vim

```diff
@@ -143,13 +143,36 @@ function s:BufSurfDelete(bufnr)
         return
     endif
 
-    " Remove the buffer from all window histories.
-    call filter(w:history, 'v:val !=' . a:bufnr)
-
-    " In case the current window history index is no longer valid, move it within boundaries.
-    if w:history_index >= len(w:history)
-        let w:history_index = len(w:history) - 1
-    endif
+    " Go into each window of each tab and remove the buffer from each window's history
+    for tab_info in gettabinfo()
+        for win_idx in tab_info.windows
+            let history = gettabwinvar(tab_info.tabnr, win_idx, 'history')
+            if type(history) != v:t_list
+                continue
+            endif
+            let history_index = gettabwinvar(tab_info.tabnr, win_idx, 'history_index')
+
+            call filter(history, 'v:val != ' . a:bufnr)
+            " Remove duplicate buffers that have been made adjacent from the deletion
```

Comment:
I think you can use `uniq` for this, no?

Reply:
You are correct. Thought `uniq` removed all copies beyond the first, not just adjacent copies.